test/functional/bundleadd/add-rc/test.bats

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/env bats
+
+load "../../swupdlib"
+
+t1_hash="24d8955d9952c3fcb2241b0f8d225205a5861cec9757b3a075d34810da9b08af"
+t2_hash="e6d85023c5e619eb43d5cfbfdbdec784afef5a82ffa54e8c93bda3e0883360a3"
+
+setup() {
+  clean_test_dir
+  create_manifest_tar 10 MoM
+  sign_manifest_mom 10
+  create_manifest_tar 10 os-core
+  create_manifest_tar 10 test-bundle1
+  create_manifest_tar 10 test-bundle2
+  chown_root "$DIR/web-dir/10/staged/$t1_hash"
+  chown_root "$DIR/web-dir/10/staged/$t2_hash"
+  tar -C "$DIR/web-dir/10" -cf "$DIR/web-dir/10/pack-test-bundle1-from-0.tar" --exclude=staged/$t1_hash/* staged/$t1_hash
+  tar -C "$DIR/web-dir/10" -cf "$DIR/web-dir/10/pack-test-bundle2-from-0.tar" staged/$t2_hash
+}
+
+teardown() {
+  clean_tars 10
+  revert_chown_root "$DIR/web-dir/10/staged/$t1_hash"
+  revert_chown_root "$DIR/web-dir/10/staged/$t2_hash"
+  revert_chown_root "$DIR/web-dir/10/Manifest.os-core"
+  revert_chown_root "$DIR/web-dir/10/Manifest.test-bundle1"
+  revert_chown_root "$DIR/web-dir/10/Manifest.test-bundle2"
+  sudo rmdir "$DIR/target-dir/usr/bin/"
+  sudo rm "$DIR/target-dir/usr/foo"
+}
+
+# 8 combinations of rc to check, but skip all off
+# bad name on/off (1)
+# existing name on/off (2)
+# new name on/off (4)
+# Do it in order 1 4 3 2 5 7 6
+
+@test "bundle-add returncodes part 1" {
```

Comment:
Is there a part 2? :) These might be better split up into separate @test "<>" sections, but this is ok for now.